docs/api_proposal.md

```diff
@@ -0,0 +1,146 @@
+
+See [API example doc](https://github.com/DINA-Web/taxonomy/blob/master/docs/api_examples.md) for notes on non-DINA taxonomy APIs
+
+## Assumptions
+
+* taxonomy = hierarchy = classification = checklist
+* there will be some canonical root URL for the API, format to be determined
+* the API will be versioned, and we may deprecate older versions in the future
+* exact format / fields of result TBD; examples below are just for discussion about general form of result for each method
+* taxonomic names are not guaranteed to be unique
+  * therefore, cannot safely perform CRUD options based on name strings without disambiguation
+* one goal is to harmonize with existing taxonomy APIs (improved interoperability with other services and future modules)
+
+## Methods
+
+Starting with GET methods. At a high level, we need methods to
+1. Search for taxa based on name strings (taxonomic name resolution)
+1. Search taxa based on some parameter
+1. Get a specific  taxon
+1. Return information about the taxonomic hierarchy / relationships between taxa.
+
+Here is a proposed list of key GET methods:
+
+### /tnrs
+Method for taxonomic name reconciliation; returns potentially matching taxa (exact or fuzzy matching; uses information about synonyms)
+
+`/tnrs?name="Genus speices"`
+
+returns:
+```
+{
+  "numberMatches" : 3,
+  "bestMatch" : {
+    "id" : 3,
+    "canonicalName" : "Genus species",
+    "matchScore" : 0.8
+    },
+  "alternateMatches" : [
+    { _info about next match here_ },
+    { _info about next match here_ }
+  ]
+}
+```
+
+### /autocomplete
+Return a list of potentially matching taxa based on given prefix; for UI search boxes
+
+`/autocomplete?prefix="Gen"`
+
+Return format same as [tnrs](#tnrs).
+
+### /search
+Search taxa based on a variety of parameters (rank, author, year, etc)  
+
+`/search?author="Linnaeus"&rank="order"`
+
+Return format same as [tnrs](#tnrs).
+
+### /taxon
+Return a specific taxon resource; takes ID as parameter
+
+  `/taxon/{id}`
+
+returns:
+```
+{
+  "id" : id,
+  "canonicalName" : "Genus species",
+  "scientificName" : "Genus species (author, year)",
+  "synonyms" : ["synonym1", "synonymn2"],
+  "rank" : "genus"
+}
+```
+
+### /about
+Return information about a taxonomic hierarchy.
+
+```
+{
+  "taxonomyId" : id,
+  "lastEdited" : "2018-12-14",
+  "numberTaxa" : 96323,
+  "rootTaxonId" : 2,
+  "rootTaxonName" : "Life"
+}
+```
+
+### /mrca
+
+Return the most recent common ancestor of two or more taxa
+
+  `/mrca?taxa=["Genus sp1", "Genus sp2"]`
+
+returns:
+```
+{
+  "mrca" : {
+    "id" : id,
+    "canonicalName" : "Genus species",
+    "scientificName" : "Genus species (author, year)",
+    "synonyms" : ["synonym1", "synonymn2"],
+    "rank" : "genus"
+  }
+}
+```
+
+### /subtree
+Return the subtree descended from a taxon
+
+`/subtree?taxon=id`
+
+Returns json with subtree, limited and / or paginated if large number of levels / children.
+
+## Questions
+
+* Will a single DINA system store more than one taxonomy?
```

Comment:
More than one taxonomy. User authentication not part of API.Done using headers by any number of mechanisms.

Reply:
I am confused about what user authentication has to do with disambiguating between multiple taxonomies.